content/challenges/easy/9 - Factorial of a Positive Integer/func.js

```diff
@@ -12,5 +12,16 @@
  * @returns {number}
  */
 module.exports = (num) => {
-   // Your code goes here
+   if (num < 0) {
+      throw new Error("Cannot Pass in a Negative Number");
+   } else if (num === 0) {
+      return 1;
+   } else { 
+      let factorial = 1;
+      while (num > 0) {
+         factorial *= num;
+         num--;
+      }
+      return factorial;
+   }
```

Comment:
take a crack at doing a factorial using `recursion`. Good place to introduce the concept!

[good place to get started](https://stackoverflow.com/questions/43426769/javascript-factorial-with-recursion)